### PR TITLE
Added UT for powervs_cluster delete functions

### PR DESCRIPTION
**Pull request by KeerthanaAP**
**What this PR does / why we need it**:
Adds UT for DeleteLoadbalancer,DeleteTransitGateway,DeleteVPC,DeleteVPCSubnet,DeleteVPCSecurityGroups functions

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/1818

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: cloud/scope/powervs_cluster_test.go [hunk 1]
```diff
@@ -3832,7 +3832,659 @@ func TestCreateVPCSubnet(t *testing.T) {
...
+		setup(t)
+		t.Cleanup(teardown)
+		clusterScope := powervsClusterScope()
+		mockVpc.EXPECT().GetLoadBalancer(gomock.Any()).Return(nil, &core.DetailedResponse{StatusCode: 404}, errors.New("not found"))
+		clusterScope.IBMVPCClient = mockVpc
+		requeue, err := clusterScope.DeleteLoadBalancer()
+		g.Expect(err).To(BeNil())
+		g.Expect(requeue).To(BeFalse())
+	})
+	t.Run("When IBMVPCClient DeleteLoadBalancer returns error", func(*testing.T) {
```
Comment (Karthik-K-N) [inline cloud/scope/powervs_cluster_test.go:3879]:
```suggestion
	t.Run("Whe DeleteLoadBalancer returns error", func(*testing.T) {
```

Comment (Karthik-K-N) [reply cloud/scope/powervs_cluster_test.go:3879]: I think we can remove IBMVPCClient here and at other relevant places.

**Review by Karthik-K-N — COMMENTED** [review]
Overall LGTM, Thanks for submitting the PR in quick time.
Few changes in test case names to remove VPCClient as I feel it not necessary.

/assign @Amulyam24

File: cloud/scope/powervs_cluster_test.go (hunk 1 above)
Comment (KeerthanaAP) [reply cloud/scope/powervs_cluster_test.go]: Done.Removed it.

**Review by Amulyam24 — COMMENTED** [review]
We support creating multiple load balancers and subnets, it would be good if you can add test cases about them which can cover scenarios such as
1. one LB is created by controller and other is not
2. Case for 3 LBs

**Comment by KeerthanaAP** [comment]
> We support creating multiple load balancers and subnets, it would be good if you can add test cases about them which can cover scenarios such as
> 
> 1. one LB is created by controller and other is not
> 2. Case for 3 LBs

@Amulyam24  Added the below 2 cases (for LoadBalancer,SecurityGroups,VPCSubnets) 
1.  Deleting multiple LBs 
2. One LB is not created by controller

**Review by Karthik-K-N — COMMENTED** [review]
/lgtm

**Pull request merged**